src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -990,19 +1057,1051 @@ partial class C
+                //     public abstract partial int P2 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberCannotBeAbstract, "P2").WithLocation(5, 33),
+                // (5,38): error CS0500: 'C.P2.get' cannot declare a body because it is marked abstract
+                //     public abstract partial int P2 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_AbstractHasBody, "get").WithArguments("C.P2.get").WithLocation(5, 38),
+                // (5,49): error CS0500: 'C.P2.set' cannot declare a body because it is marked abstract
+                //     public abstract partial int P2 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_AbstractHasBody, "set").WithArguments("C.P2.set").WithLocation(5, 49),
+                // (7,33): error CS0750: A partial member cannot have the 'abstract' modifier
+                //     public abstract partial int P3 { get; set; }
+                Diagnostic(ErrorCode.ERR_PartialMemberCannotBeAbstract, "P3").WithLocation(7, 33),
+                // (8,38): error CS0500: 'C.P3.get' cannot declare a body because it is marked abstract
+                //     public abstract partial int P3 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_AbstractHasBody, "get").WithArguments("C.P3.get").WithLocation(8, 38),
+                // (8,49): error CS0500: 'C.P3.set' cannot declare a body because it is marked abstract
+                //     public abstract partial int P3 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_AbstractHasBody, "set").WithArguments("C.P3.set").WithLocation(8, 49),
+                // (10,33): error CS0750: A partial member cannot have the 'abstract' modifier
+                //     public abstract partial int P4 { get; set; }
+                Diagnostic(ErrorCode.ERR_PartialMemberCannotBeAbstract, "P4").WithLocation(10, 33),
+                // (11,36): error CS0029: Cannot implicitly convert type 'string' to 'int'
+                //     public partial int P4 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""""").WithArguments("string", "int").WithLocation(11, 36),
+                // (14,38): error CS0500: 'C.P5.get' cannot declare a body because it is marked abstract
+                //     public abstract partial int P5 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_AbstractHasBody, "get").WithArguments("C.P5.get").WithLocation(14, 38),
+                // (14,49): error CS0500: 'C.P5.set' cannot declare a body because it is marked abstract
+                //     public abstract partial int P5 { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_AbstractHasBody, "set").WithArguments("C.P5.set").WithLocation(14, 49));
+        }
+
+        [Fact]
+        public void Required_Errors()
+        {
+            var source = """
+                partial class C
+                {
+                    public required partial string P1 { get; set; }
```

Comment:
Please separate out the happy case and verify that the property is treated as required as intended